plugins/zulip/config/static_config.py

```diff
@@ -0,0 +1,113 @@
+"""Configurations fixed for plugin"""
+import logging
+import os
+from dataclasses import dataclass
+from typing import (
+    Dict,
+    List,
+    Type as __Type,
+    TypeVar as __TypeVar
+)
+
+import yaml
+
+from plugins.zulip.helpers.loader import special_loader
+
+LOGGER = logging.getLogger(__name__)
+LOGGER.setLevel(logging.INFO)
+
+T = __TypeVar("T")
+
+
+class Resources:
+    """Resource retrieval from dedicated resource directory.
+
+    Example:
+        ``resources = Resources(__file__)``
+
+        ``resources["my_file.txt"]``
+
+            will have return of:
+
+        ``C:\\\\Path\\to\\python_file\\resources\\my_file.txt``
+    """
+
+    def __init__(self, content_root: str, resources_dir: str = "resources", create_resource_root=False):
+        """Initialize resource retrieval, normally `content_root` should be `__file__`"""
+        if os.path.isfile(content_root):
+            content_root = os.path.dirname(content_root)
+        self.resource_root = os.path.abspath(f"{content_root}/{resources_dir}")
+        if create_resource_root and not os.path.exists(self.resource_root):
+            LOGGER.warning("No resource root exists at %s, directory will be created", content_root)
+            os.makedirs(self.resource_root, exist_ok=True)
+
+    def __repr__(self):
+        return f"{type(self).__name__} at {self.resource_root}"
+
+    def __getitem__(self, resource_name):
+        """Return path to resource by given name. If given path is absolute, return if without change"""
+        if os.path.isabs(resource_name):
+            return resource_name
+        return os.path.abspath(f"{self.resource_root}/{resource_name}")
+
+
+def load(stream, as_type: __Type[T]) -> T:
```

Comment:
this is going too far from the python POV. Do you really want to make c++ out of python?